ddtrace/tracer/textmap.go

```diff
@@ -261,9 +268,46 @@ func (p *propagator) injectTextMap(spanCtx ddtrace.SpanContext, writer TextMapWr
 	for k, v := range ctx.baggage {
 		writer.Set(p.cfg.BaggagePrefix+k, v)
 	}
+	if p.cfg.MaxTagsHeaderLen <= 0 {
+		return nil
+	}
+	// propagate trace tags
+	sb := p.formatPropagatingTags(ctx)
+	if sb.Len() > 0 {
+		writer.Set(traceTagsHeader, sb.String())
+	}
 	return nil
 }
 
+func (p *propagator) formatPropagatingTags(ctx *spanContext) strings.Builder {
+	var sb strings.Builder
+	if ctx.trace == nil {
+		return sb
+	}
+	ctx.trace.mu.Lock()
+	defer ctx.trace.mu.Unlock()
+	for k, v := range ctx.trace.propagatingTags {
+		if err := isValidPropagatableTraceTag(k, v); err != nil {
+			log.Warn("won't propagate tag '%s' (err: %s)", k, err.Error())
+			ctx.trace.setTag(propagationErrorKey, "encoding_error")
+			continue
+		}
+		if sb.Len()+len(k)+len(v) > p.cfg.MaxTagsHeaderLen {
+			sb.Reset()
+			log.Warn("won't propagate trace tags (err: max trace tags header len (%d) reached)", p.cfg.MaxTagsHeaderLen)
```

Comment:
```suggestion
			log.Warn("Won't propagate trace tags: maximum trace tags header length (%d) reached.", p.cfg.MaxTagsHeaderLen)
```
Please remember that this is for humans. Log messages should be capitalised.